WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -3993,61 +4207,99 @@ public void onPostChanged(OnPostChanged event) {
+                                    })
+                                    .show();
+                        }
 
-                if (mIsDiscardingChanges) {
-                    mIsDiscardingChanges = false;
-                    mPost = mPostStore.getPostByLocalPostId(mPost.getId());
-                    mDispatcher.dispatch(PostActionBuilder.newUpdatePostAction(mPost));
-                    mIsUpdatingPost = true;
+                        updatePostLoadingAndDialogState(PostLoadingState.NONE);
+                        break;
+                    case DISCARDING:
+                        mPostLoadingState = PostLoadingState.RELOADING;
+                        mPost = mPostStore.getPostByLocalPostId(mPost.getId());
+                        mDispatcher.dispatch(PostActionBuilder.newUpdatePostAction(mPost));
+                        break;
                 }
             } else {
-                if (mIsDiscardingChanges) {
+                if (mPostLoadingState == PostLoadingState.DISCARDING) {
                     showDialogError(R.string.local_changes_discarding_error, R.string.contact_support,
                             TAG_DISCARDING_CHANGES_ERROR_DIALOG,
                             true);
                 }
 
-                mIsDiscardingChanges = false;
-                mIsUpdatingPost = false;
-                showDialogProgress(false);
+                updatePostLoadingAndDialogState(PostLoadingState.NONE);
                 AppLog.e(AppLog.T.POSTS, "UPDATE_POST failed: " + event.error.type + " - " + event.error.message);
             }
+        } else if (event.causeOfChange instanceof CauseOfOnPostChanged.RemoteAutoSavePost) {
+            if (!event.isError()) {
+                mPost = mPostStore.getPostByLocalPostId(mPost.getId());
+            } else {
+                AppLog.e(T.POSTS, "REMOTE_AUTO_SAVE_POST failed: " + event.error.type + " - " + event.error.message);
+                updatePostLoadingAndDialogState(PostLoadingState.REMOTE_AUTO_SAVE_PREVIEW_ERROR, mPost);
+            }
+            RemotePostPayload payload = new RemotePostPayload(mPost, mSite);
+            mDispatcher.dispatch(UploadActionBuilder.newPushedPostAction(payload));
         }
     }
 
+
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onPostUploaded(OnPostUploaded event) {
         final PostModel post = event.post;
         if (post != null && post.getId() == mPost.getId()) {
-            View snackbarAttachView = findViewById(R.id.editor_activity);
-            UploadUtils.onPostUploadedSnackbarHandler(this, snackbarAttachView, event.isError(), post,
-                    event.isError() ? event.error.message : null, getSite(), mDispatcher);
-            if (!event.isError()) {
-                mPost = post;
-                mIsNewPost = false;
-                invalidateOptionsMenu();
+            if (!(mPostLoadingState == PostLoadingState.UPLOADING_FOR_PREVIEW)
```

Comment:
This method is very hard to read and quite hard to understand. I know `EditPostActivity` is not a model of good practices, but can you split this into smaller methods and add some comments to explain what's going on?
Some comments like: "When we get a post update originating from the preview feature: open the internal browse for this post".

Reply:
You are absolutely right, I have to admit I did not get it while coding but it is not clear to read and understand. I tried to improve it here 42d13f43b94884e1457fbbe4aee99a757a470133. Hope it is better.

Reply:
Looks better thanks!